tests/unit/test_utils.py

```diff
@@ -486,6 +493,11 @@ def test_auto_decode_no_preferred_encoding(self):
                 ret = auto_decode(data.encode(sys.getdefaultencoding()))
         assert ret == data
 
+    @pytest.mark.parametrize('encoding', (encoding for bom, encoding in BOMS))
```

Comment:
It would be better to call `list` or `tuple` here rather than passing a raw generator expression. (It's not clear how many times pytest iterates over the expression, and pytest doesn't say generator expressions are supported here: https://docs.pytest.org/en/latest/reference.html#pytest-mark-parametrize )